File: R/raindrops/raindrops.R
raindrops <- function(number) {

  stopifnot(is.numeric(number))

  fac_str_pairs <- data.frame(
    fac              = c(3, 5, 7),
    str              = c("Pling", "Plang", "Plong"),
    stringsAsFactors = FALSE
  )

  is_fac <- ((number %% fac_str_pairs$fac) == 0)
  `if`(
    any(is_fac),
    paste(ifelse(is_fac, fac_str_pairs$str, ""), collapse = ""),
    as.character(number)
  )
}

File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(def drop-sounds
  {3 "Pling"
   5 "Plang"
   7 "Plong"})

(defn convert [n]
  (let [is-fac-3 (= 0 (mod n 3))
        is-fac-5 (= 0 (mod n 5))
        is-fac-7 (= 0 (mod n 7))
        is-fac-any (or is-fac-3 is-fac-5 is-fac-7)]
    (if is-fac-any
      (str (if is-fac-3 (get drop-sounds 3) "")
           (if is-fac-5 (get drop-sounds 5) "")
           (if is-fac-7 (get drop-sounds 7) ""))
      (str n))))
